src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,228 @@
+                (fraction with respect to camera median value)""",
+    ).tag(config=True)
+    image_std_cut_outliers = List(
+        [-3, 3],
+        help="""Interval (number of std) of accepted image standard deviation \\
+                around camera median value""",
+    ).tag(config=True)
+
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        """
+        Base component to handle the extraction of the statistics
+        from a chunk of charges and pulse times (images).
+
+        Parameters
+        ----------
+        kwargs
+        """
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+
+    def __call__(
+        self,
+        dl1_table,
+        masked_pixels_of_sample=None,
+        chunk_shift=None,
+        col_name="image",
+    ) -> list:
+        """
+        Call the relevant functions to extract the statistics
+        for the particular extractor.
+
+        Parameters
+        ----------
+        dl1_table : astropy.table.QTable
+            DL1 table with images of shape (n_images, n_channels, n_pix) and timestamps of shape (n_images, ) stored in an astropy QTable
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels of shape (n_pix, ) that are not available for processing
+        chunk_shift : int
+            number of samples to shift the extraction chunk
+        col_name : string
+            column name in the DL1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and extraction chunks
+        """
+
+        # Check if the length of the dl1 table is greater or equal than the size of the chunk.
+        if len(dl1_table[col_name]) < self.chunk_size:
+            raise ValueError(
+                f"The length of the DL1 table '{len(dl1_table[col_name])}' must be greater or equal than the size of the chunk '{self.chunk_size}'."
+            )
+        # If no chunk_shift is provided, the chunk_shift is set to self.chunk_size
+        # meaning that the extraction chunks are not overlapping.
+        if chunk_shift is None:
+            chunk_shift = self.chunk_size
+
+        # Function to split table data into appropriated chunks
+        def _get_chunks(col_name):
+            return [
+                (
+                    dl1_table[col_name].data[i : i + self.chunk_size]
+                    if i + self.chunk_size <= len(dl1_table[col_name])
+                    else dl1_table[col_name].data[
+                        len(dl1_table[col_name].data) - self.chunk_size : len(
+                            dl1_table[col_name].data
+                        )
+                    ]
+                )
+                for i in range(0, len(dl1_table[col_name].data), chunk_shift)
+            ]
+
+        # Get the chunks for the timestamps and selected column name
+        time_chunks = _get_chunks("time")
+        image_chunks = _get_chunks(col_name)
```

Comment:
Actually, the operation `dl1_table[col_name].data` works only for `<class 'astropy.table.column.Column'>` objects. However, `dl1_table["time"]` is a `<class 'astropy.time.core.Time'>` object. My suggestion would be:
```
        # Function to split table data into appropriated chunks
        def _get_chunks(dl1_table_data):
            return [
                (
                    dl1_table_data[i : i + self.chunk_size]
                    if i + self.chunk_size <= len(dl1_table_data)
                    else dl1_table_data[
                        len(dl1_table_data) - self.chunk_size : len(
                        dl1_table_data
                        )
                    ]
                )
                for i in range(0, len(dl1_table_data), chunk_shift)
            ]

        # Get the chunks for the timestamps and selected column name
        time_chunks = _get_chunks(dl1_table["time"])
        image_chunks = _get_chunks(dl1_table[col_name].data)
```
What do you think, @mexanick @Hckjs?

Reply:
looks aok, but please also add relevant tests

Reply:
Resolved by [e8497fd](https://github.com/cta-observatory/ctapipe/pull/2554/commits/e8497fde9386d17644bdf8bc02deba736658cdf3).